### ISSUE TITLE
Problems when unable to end on a delimiter 

### ISSUE THREAD

**Issue by erik-stephens** [terascope/chunked-file-reader#6]
I'm seeing a couple of issues:

1. The ["2x avg size" assumption](https://github.com/terascope/chunked-file-reader/blob/abc87519dc3a5001b3c553d4f53d8cea801bb918/index.js#L42) is not safe to make.  Thoughts on how to handle the case when the last bit + margin doesn't include the delimiter?  Keep reading until there is a delimiter?

1. When not enough is read at the end, the formatter fails too quietly / no error is thrown.  It manifests itself downstream as `data=null`.

**Comment by erik-stephens** [comment]
After some discussion, we think ok to expand the margin until we read a delimiter.  We considered some edge cases:

1. There is little risk of reading too far since we [only consider up to first delimiter in the margin](https://github.com/terascope/chunked-file-reader/blob/abc87519dc3a5001b3c553d4f53d8cea801bb918/index.js#L19), unless...

1. The record is larger than chunk size.  Handling this case not worth the complexity, so going to require users to set their chunk sizes large enough.

Regarding the 2nd issue, formatters should catch at each record to avoid one bad apple spoiling the bunch:
```javascript
> i = Promise.map(['{"a":1}', 'null', '{"yeee":boye}'], r => { try {return JSON.parse(r);} catch (e) {return 'NOPE';} }).filter(i => i).catch(console.error)
Promise {
  _bitField: 0,
  _fulfillmentHandler0: undefined,
  _rejectionHandler0: undefined,
  _promise0: undefined,
  _receiver0: undefined }
> i.value()
[ { a: 1 }, 'NOPE' ]
> i = Promise.map(['{"a":1}', 'null', '{"yeee":boye}'], r => JSON.parse(r)).filter(i => i).catch(console.error)
Promise {
  _bitField: 0,
  _fulfillmentHandler0: undefined,
  _rejectionHandler0: undefined,
  _promise0: undefined,
  _receiver0: undefined }
> SyntaxError: Unexpected token b in JSON at position 8
    at JSON.parse (<anonymous>)
    at Promise.map.r (repl:1:65)
    at bound (domain.js:301:14)
    at runBound (domain.js:314:12)
    at tryCatcher (/Users/estephens/src/logbus/node_modules/bluebird/js/release/util.js:16:23)
    at MappingPromiseArray._promiseFulfilled (/Users/estephens/src/logbus/node_modules/bluebird/js/release/map.js:61:38)
    at MappingPromiseArray.PromiseArray._iterate (/Users/estephens/src/logbus/node_modules/bluebird/js/release/promise_array.js:114:31)
    at MappingPromiseArray.init (/Users/estephens/src/logbus/node_modules/bluebird/js/release/promise_array.js:78:10)
    at MappingPromiseArray._asyncInit (/Users/estephens/src/logbus/node_modules/bluebird/js/release/map.js:30:10)
    at Async._drainQueue (/Users/estephens/src/logbus/node_modules/bluebird/js/release/async.js:138:12)
    at Async._drainQueues (/Users/estephens/src/logbus/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/Users/estephens/src/logbus/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:810:20)
    at tryOnImmediate (timers.js:768:5)
    at processImmediate [as _immediateCallback] (timers.js:745:5)
i.value()
undefined
```

**Comment by erik-stephens** [comment]
Fixed in #7
